### PR TITLE
[MM-31497] - Prevent end users from inviting people and exceeding the free tier limits

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Adds functionality to prevent non-admins from inviting people and exceeding the free tier limits

#### Ticket Link
[MM-31497](https://mattermost.atlassian.net/browse/MM-31497?atlOrigin=eyJpIjoiMDJhMDQwOWM1ODZhNDMzM2IyNDE5ZjAxNDM1OTg4Y2UiLCJwIjoiaiJ9)

#### Related Pull Requests

- Has server changes (https://github.com/mattermost/mattermost-server/pull/16786)
- Has redux changes (https://github.com/mattermost/mattermost-redux/pull/1343)

### CODE REVIEW

File: components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx
Comment (marianunez) [inline components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx:42]: Given that we don't need the subscription here anymore, maybe we don't need the HOC `withGetCloudSubscription` ?

File: components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx
Comment (marianunez) [inline components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx:45]: Same here regarding the need of `withGetCloudSubscription`

File: components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx
Comment (marianunez) [inline components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx:184]: We should check here if they are on cloud license before we try to get the `subscriptionStats`

**Review by marianunez — CHANGES_REQUESTED** [review]

**Review by marianunez — COMMENTED** [review]
@AGMETEOR redux hash needs to be updated to the latest where you have your endpoint defined. It's erroring out in the test server

File: components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx
Comment (nickmisasi) [reply components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx:184]: Isn't that what the getWithCloudSubscription HOC does?

Comment (marianunez) [reply components/invitation_modal/invitation_modal_members_step/invitation_modal_members_step.jsx:184]: Well `getWithCloudSubscription` does the actual get of the `subscription`, this is a new endpoint: https://github.com/mattermost/mattermost-server/pull/16786

File: components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx
Comment (AGMETEOR) [reply components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx:42]: That's true.

**Comment by AGMETEOR** [comment]
> @AGMETEOR redux hash needs to be updated to the latest where you have your endpoint defined. It's erroring out in the test server

Was still waiting on the PR there to get merged.

File: package.json
Comment (marianunez) [inline package.json:42]: The package-lock.json also needs to be updated to the corresponding redux hash for it to load properly.

**Review by marianunez — APPROVED** [review]
LGTM, just a comment on the redux hash missing.

File: components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx
Comment (marianunez) [inline components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx:180]: Missed in the initial pass that we should only call `getSubscriptionStats` if it's in cloud. Also, if not in cloud, could `props.subscriptionStats` be null? If so, let's add a null check when necessary to avoid issues

**Review by nickmisasi — APPROVED** [review]
+1 on Maria's latest comment, but otherwise LGTM.

**Review by marianunez — CHANGES_REQUESTED** [review]
Changing to request changes in order to complete the cloud check before getting subscription stats

**Review by stevemudie — APPROVED** [review]
LGTM

File: components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx
Comment (AGMETEOR) [reply components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.jsx]: Hi @marianunez thanks for the review. I have updated the code to make the checks. Here's the redux PR https://github.com/mattermost/mattermost-redux/pull/1358 that needs approval before this one can be merged.

**Review by marianunez — APPROVED** [review]
Thanks @AGMETEOR!

**Pull request merged**
